src/Contract/Operation/FlatMapable.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+/**
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace loophp\collection\Contract\Operation;
+
+use Iterator;
+use loophp\collection\Contract\Collection;
+
+/**
+ * @template TKey
+ * @template T
+ */
+interface FlatMapable
+{
+    /**
+     * Transform the collection through a callback and flatten one level.
+     *
+     * @template V
+     *
+     * @param callable(T, TKey, Iterator<TKey, T>): V $callback
```

Comment:
Shouldn't this be:

```suggestion
     * @param callable(T, TKey, Iterator<TKey, T>): iterable<IterableKey, IterableValue> $callback
```

Reply:
yeah the types are not yet right, I will get them right when I add the static analysis check :)